sycl/doc/extensions/experimental/sycl_ext_oneapi_kernel_compiler.asciidoc

```diff
@@ -461,51 +597,301 @@ is `name`.
+
+sycl::kernel_bundle<sycl::bundle_state::executable> kb = syclex::build(kb_src,
+  syclex::properties{syclex::registered_kernel_names{{"bartmpl<float>", "bartmpl<int>"}});
+
+sycl::kernel k_float = kb.ext_oneapi_get_kernel("bartmpl<float>");
+sycl::kernel k_int = kb.ext_oneapi_get_kernel("bartmpl<int>");
+```
+
+
+== Examples
+
+=== Simple example
 
 The following example demonstrates how a SYCL application can define a kernel
 as a string and then compile and launch it.
 
 ```
 #include <sycl/sycl.hpp>
 namespace syclex = sycl::ext::oneapi::experimental;
+static constexpr size_t NUM = 1024;
 
 int main() {
   sycl::queue q;
 
-  // The source code for one or more kernels, defined in one of
-  // the supported source languages.
+  // The source code for a kernel, defined as a SYCL "free function kernel".
   std::string source = R"""(
-    /* language specific kernel source code */
+    #include <sycl/sycl.hpp>
+    namespace syclex = sycl::ext::oneapi::experimental;
+
+    extern "C"
+    SYCL_EXT_ONEAPI_FUNCTION_PROPERTY((syclex::range_kernel<1>))
+    void iota(float start, float *ptr) {
+      size_t id = syclex::this_kernel::get_id();
+      ptr[id] = start + static_cast<float>(id);
+    }
   )""";
 
-  // Create a kernel bundle in "source" state.  The "some-language" is
-  // a stand-in for the enumerator telling which source language is used.
+  // Create a kernel bundle in "source" state.
   sycl::kernel_bundle<sycl::bundle_state::ext_oneapi_source> kb_src =
     syclex::create_kernel_bundle_from_source(
       q.get_context(),
-      syclex::source_language::/*some-language*/,
+      syclex::source_language::sycl,
       source);
 
+  // Compile the kernel.  There is no need to use the "registered_kernel_names"
+  // property because the kernel is declared extern "C".
   sycl::kernel_bundle<sycl::bundle_state::executable> kb_exe =
     syclex::build(kb_src);
 
-  // Get the kernel via its name.  The "kernel-name" is a stand-in for the
-  // actual kernel name in the source string.
-  sycl::kernel k = kb_exe.ext_oneapi_get_kernel("kernel-name");
+  // Get the kernel via its compiler-generated name.
+  sycl::kernel iota = kb_exe.ext_oneapi_get_kernel("iota");
 
+  float *ptr = sycl::malloc_shared<float>(NUM, q);
   q.submit([&](sycl::handler &cgh) {
-    // Any arguments for the kernel must be set manually.
-    cgh.set_args(/*...*/);
+    // Set the values of the kernel arguments.
+    cgh.set_args(3.14f, ptr);
 
-    // Launch the kernel according to its type.
-    // This assumes a simple "range" kernel.
-    cgh.parallel_for(sycl::range{1024}, k);
-  });
+    // Launch the kernel according to its type, in this case a simple
+    // "range" kernel.
+    cgh.parallel_for({NUM}, iota);
+  }).wait();
 }
 ```
+
+=== Disambiguating overloaded kernel functions
+
+This example demonstrates how to use the `registered_kernel_names` property to
+disambiguate a kernel function that has several overloads.
+
+```
```

Comment:
Should we format as C++ all the code listings in the proposal?

Reply:
Also yes.  Done in 5f856625a084fecdd9cb190b19dce422000b37f8.